connman/src/agent.c

```diff
@@ -257,6 +257,32 @@ int connman_agent_queue_message(void *user_context,
 	return err;
 }
 
+bool connman_agent_queue_search(void *user_context, void *agent_data)
+{
+	struct connman_agent *agent = agent_data;
+	struct connman_agent_request *queue_data;
+	int len;
+	int i;
+
+	DBG("user context %p agent %p", user_context, agent);
+
+	if (!agent || !user_context)
+		return false;
+
+	len = g_list_length(agent->queue);
+
+	for (i = 0; i < len; i++) {
+		queue_data = g_list_nth_data(agent->queue, i);
```

Comment:
This is now O(N²) algorithm. Just thinking that to me it would make sense to iterate the list without using indices like below or use `g_list_find_custom()`.
```C
for (GList *elem = agent->queue; elem != NULL; elem = elem->next) {
    // check elem
}
```

Reply:
As my eyes fooled me to see `GList` as `GSList`.

Reply:
> I really did not yet think this but to get something for the test.

Ok. I didn't realize that myself.

Reply:
I think this is ok now.